specs/_features/eip7732/beacon-chain.md

```diff
@@ -0,0 +1,716 @@
+
+```python
+def process_execution_payload(state: BeaconState, 
+                              signed_envelope: SignedExecutionPayloadEnvelope, 
+                              execution_engine: ExecutionEngine, verify: bool = True) -> None:
+    # Verify signature
+    if verify: 
+        assert verify_execution_payload_envelope_signature(state, signed_envelope)
+    envelope = signed_envelope.message
+    payload = envelope.payload
+    # Cache latest block header state root
+    previous_state_root = hash_tree_root(state)
+    if state.latest_block_header.state_root == Root():
+        state.latest_block_header.state_root = previous_state_root
+    
+    # Verify consistency with the beacon block
+    assert envelope.beacon_block_root == hash_tree_root(state.latest_block_header)
+
+    # Verify consistency with the committed header
+    committed_header = state.latest_execution_payload_header
+    assert envelope.builder_index == committed_header.builder_index
+    assert committed_header.blob_kzg_commitments_root == hash_tree_root(envelope.blob_kzg_commitments)
+
+    if not envelope.payload_withheld: 
+        # Verify the withdrawals root
+        assert hash_tree_root(payload.withdrawals) == state.latest_withdrawals_root
+
+        # Verify the gas_limit
+        assert committed_header.gas_limit == payload.gas_limit
+
+        assert committed_header.block_hash == payload.block_hash 
+        # Verify consistency of the parent hash with respect to the previous execution payload
+        assert payload.parent_hash == state.latest_block_hash
+        # Verify prev_randao
+        assert payload.prev_randao == get_randao_mix(state, get_current_epoch(state))
+        # Verify timestamp
+        assert payload.timestamp == compute_timestamp_at_slot(state, state.slot)
+        # Verify commitments are under limit
+        assert len(envelope.blob_kzg_commitments) <= MAX_BLOBS_PER_BLOCK
+        # Verify the execution payload is valid
+        versioned_hashes = [kzg_commitment_to_versioned_hash(commitment) 
+                            for commitment in envelope.blob_kzg_commitments]
+        assert execution_engine.verify_and_notify_new_payload(
+            NewPayloadRequest(
+                execution_payload=payload,
+                versioned_hashes=versioned_hashes,
+                parent_beacon_block_root=state.latest_block_header.parent_root,
+            )
+        )
+
+        # Process Electra operations
+        def for_ops(operations: Sequence[Any], fn: Callable[[BeaconState, Any], None]) -> None:
+            for operation in operations:
+                fn(state, operation)
+
+        for_ops(payload.deposit_requests, process_deposit_request)
+        for_ops(payload.withdrawal_requests, process_withdrawal_request)
+        for_ops(payload, process_consolidation_request)
```

Comment:
This should probably be `payload.consolidation_requests`

Reply:
thanks!